src/EnergyPlus/ScheduleManager.cc

```diff
@@ -84,51 +84,61 @@ namespace ScheduleManager {
     //                      J. Glazer January 2005 -- added Schedule:File
     //                      Michael Wetter February 2010 -- added Schedule for external Interface
     //                      L Lawrie - October 2012 - added sub-hourly option for Schedule:File
-    //       RE-ENGINEERED  na
 
     // PURPOSE OF THIS MODULE:
     // To provide the capabilities of getting the schedule data from the input,
     // validating it, and storing it in such a manner that the schedule manager
     // can provide the scheduling value needs for the simulation.
 
-    // METHODOLOGY EMPLOYED:
-    // na
-
     // REFERENCES:
     // Proposal for Schedule Manager in EnergyPlus (Rick Strand)
 
     // MODULE PARAMETER DEFINITIONS
     static constexpr std::string_view BlankString;
-    Array1D_string ValidDayTypes(MaxDayTypes,
-                                 {"Sunday",
-                                  "Monday",
-                                  "Tuesday",
-                                  "Wednesday",
-                                  "Thursday",
-                                  "Friday",
-                                  "Saturday",
-                                  "Holiday",
-                                  "SummerDesignDay",
-                                  "WinterDesignDay",
-                                  "CustomDay1",
-                                  "CustomDay2"});
-
-    int constexpr NumScheduleTypeLimitUnitTypes(14);
-    Array1D_string ScheduleTypeLimitUnitTypes(NumScheduleTypeLimitUnitTypes,
-                                              {"Dimensionless",
-                                               "Temperature",
-                                               "DeltaTemperature",
-                                               "PrecipitationRate",
-                                               "Angle",
-                                               "ConvectionCoefficient",
-                                               "ActivityLevel",
-                                               "Velocity",
-                                               "Capacity",
-                                               "Power",
-                                               "Availability",
-                                               "Percent",
-                                               "Control",
-                                               "Mode"});
+    // Day types are 1-based for EMS and output and other uses, so add a dummy
+    constexpr std::array<std::string_view, static_cast<int>(DayType::Num)> dayTypeNames{"dummy",
+                                                                                        "Sunday",
```

Comment:
Single set of arrays for dayTypeNames and DayTypeNamesUC. Replaces a whole bunch of other string arrays scattered throughout the code.